gobblin-runtime/src/main/java/gobblin/runtime/TaskState.java

```diff
@@ -161,18 +168,19 @@ public void setTaskDuration(long duration) {
    * Update record-level metrics.
    *
    * @param recordsWritten number of records written by the writer
+   * @param branchIndex fork branch index
    */
-  public void updateRecordMetrics(long recordsWritten) {
+  public void updateRecordMetrics(long recordsWritten, int branchIndex) {
     JobMetrics metrics = JobMetrics.get(this.getProp(ConfigurationKeys.JOB_NAME_KEY), this.jobId);
+    String forkBranchId = ForkOperatorUtils.getForkId(this.taskId, branchIndex);
 
```

Comment:
Are the lines below necessary? Can't you just do:

"metrics.getCounter(JobMetrics.metricName(JobMetrics.MetricGroup.TASK, forkBranchId, RECORDS)).inc(inc);"

My understanding is that the MRTaskStateTracker will only call this method when the task has finished. So this method only gets called once per fork, in which case you only need to increment it once.